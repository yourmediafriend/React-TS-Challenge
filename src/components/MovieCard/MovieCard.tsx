import React from "react";
import { Link } from "react-router-dom";
import MoviePoster from "../MoviePoster/MoviePoster";
import { movieDetailType } from "../../utils/types/movieDetailType";
import styled from "styled-components";
import { TMDB_IMAGESIZES } from "../../utils/constants/tmdbApi";

const LinkSC = styled(Link)`
	border: 1px solid #c7c7c7;
	text-decoration: none;
	color: #373737;
`;
const MoviePosterSC = styled(MoviePoster)`
	width: 100%;
`;

const MovieDetailSC = styled.div`
	width: 100%;
	padding: 10px;
`;

const MovieTtitleSC = styled.h3`
	font-size: 18px;
	font-weight: bold;
	line-height: 1.3em;
`;

type CardProps = Partial<movieDetailType>;

const Card = ({
	id,
	poster_path,
	original_title,
	release_date,
	title,
	popularity,
}: CardProps) => {
	const yyyy = release_date && new Date(release_date).getFullYear();

	return (
		<LinkSC to={`movie/${id}`} data-testid="movie-link">
			{poster_path && title && (
				<MoviePosterSC
					image_url={poster_path}
					alt={title}
					size={TMDB_IMAGESIZES.poster_sizes[3]}
				/>
			)}
			<MovieDetailSC>
				<MovieTtitleSC className={"title"} data-testid="movie-title">
					{title}
				</MovieTtitleSC>
				<h4 className={"releaseDate"} data-testid="movie-release-year">
					{yyyy}
				</h4>
				<p data-testid="movie-rating">{popularity}</p>
			</MovieDetailSC>
		</LinkSC>
	);
};

export default Card;
