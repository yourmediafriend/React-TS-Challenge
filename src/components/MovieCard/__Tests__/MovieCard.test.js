import * as React from "react";
import TestRenderer from "react-test-renderer";
import { MemoryRouter as Router } from "react-router-dom";
import {
	render,
	screen,
	fireEvent,
	cleanup,
	done,
} from "@testing-library/react";
import MovieCard from "../MovieCard";
import { movieDetailShort as detail } from "../../../__Tests__/__mockData.json";

beforeEach(() => {
	render(
		<Router>
			<MovieCard
				id={detail.id}
				poster_path={detail.poster_path}
				original_title={detail.original_title}
				title={detail.title}
				release_date={detail.release_date}
				popularity={detail.popularity}
			/>
		</Router>
	);
});

afterEach(cleanup);

describe(`Movie Card`, () => {
	it("Has Link to Move Detail Page", () => {
		expect(screen.getByTestId("movie-link")).toBeTruthy();
	});

	it("Has Movie Title", () => {
		expect(screen.getByTestId("movie-title")).toBeTruthy();
	});

	it("Has Movie Image", () => {
		expect(screen.getByTestId("movie-poster")).toBeTruthy();
	});

	it("Has Movie Rating", () => {
		expect(screen.getByTestId("movie-release-year")).toBeTruthy();
	});
	it("Has Movie Date", () => {
		expect(screen.getByTestId("movie-rating")).toBeTruthy();
	});

	it(`Matches snapshot`, () => {
		const tree = TestRenderer.create(
			<Router>
				<MovieCard
					id={detail.id}
					poster_path={detail.poster_path}
					original_title={detail.original_title}
					title={detail.title}
					release_date={detail.release_date}
					popularity={detail.popularity}
				/>
			</Router>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
