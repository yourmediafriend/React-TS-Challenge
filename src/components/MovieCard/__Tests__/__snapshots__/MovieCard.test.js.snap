// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Card Matches snapshot 1`] = `
<a
  className="MovieCard__LinkSC-sc-1q6eeff-0 bbvbZr"
  data-testid="movie-link"
  href="/movie/13382"
  onClick={[Function]}
>
  <img
    alt="Rudolph the Red-Nosed Reindeer"
    className="MovieCard__MoviePosterSC-sc-1q6eeff-1 bRQxcS"
    data-testid="movie-poster"
    src="https://www.themoviedb.org/t/p/w342//WvJj5TsW3lPJcbbvZtJazsFhvY.jpg"
  />
  <div
    className="MovieCard__MovieDetailSC-sc-1q6eeff-2 egIeWa"
  >
    <h3
      className="MovieCard__MovieTtitleSC-sc-1q6eeff-3 hHbbLc title"
      data-testid="movie-title"
    >
      Rudolph the Red-Nosed Reindeer
    </h3>
    <h4
      className="releaseDate"
      data-testid="movie-release-year"
    >
      1964
    </h4>
    <p
      data-testid="movie-rating"
    >
      16.869
    </p>
  </div>
</a>
`;
